lib/internal/modules/cjs/loader.js

```diff
@@ -554,6 +555,8 @@ Module._load = function(request, parent, isMain) {
   try {
     module.load(filename);
     threw = false;
+  }catch(err){
+    throw new ESM_CJS_LOAD_ERROR(err);
```

Comment:
Here's how I'd suggest doing the error adjustment here:

```suggestion
  if (err instanceof SyntaxError) {
    checkAddEsmNote(err);
  }
  throw err;
```

and then create a separate function like the following:

```js
function checkAddEsmNote (err) {
  if (err.message.startsWith('SyntaxError: Unexpected token export') || ...importRegExCheck...) {
    // we now know it is an import / export syntax error so is likely a module
    // mutate the error to include the new info -
    err.stack = "INFO" + err.stack;
  }
}
```